File: Year_2026/February/sum_of_root_to_leaf_binary_numbers.java
package February;

public class sum_of_root_to_leaf_binary_numbers {
    static void main() {

    }

    static class Solution {
        public int sumRootToLeaf(TreeNode root) {
            return solve(root, 0);
        }

        private int solve(TreeNode root, int currSum) {
            if (root == null) return 0;
            currSum <<= 1;
            currSum |= root.val;
            if (root.left == null && root.right == null) return currSum;
            return solve(root.left, currSum) + solve(root.right, currSum);
        }
    }
}

File: Year_2026/February/ruby/sum_of_root_to_leaf_binary_numbers.rb
def sum_root_to_leaf(root)
  solve(root , 0)
end

def solve(root , currSum)
  return 0 if root == nil
  currSum <<= 1
  currSum |= root.val
  return currSum if root.left == nil && root.right == nil
  solve(root.left , currSum) + solve(root.right , currSum)
end